src/main/java/com/bzabek/ticketbookingapp/service/NameAndSurnameValidator.java

```diff
@@ -0,0 +1,56 @@
+package com.bzabek.ticketbookingapp.service;
+
+
+
+import com.bzabek.ticketbookingapp.exceptions.NameSurnameException;
+
+import java.util.regex.Pattern;
+
+class NameAndSurnameValidator {
+
+    private NameAndSurnameValidator() {
+    }
+
+    public static void validate(String str) {
+        Pattern pattern = Pattern.compile("[^a-zA-ząćęłńóśźżĄĆĘŁŃÓŚŹŻ\\s]");
```

Comment:
za każdym razem kompilujesz regexa - czy to wydajne?

Reply:
Kompilacja regexa jest dosyć kosztowna. Nie jest to wydajne.
Powinienem był utworzyc statyczny skompilowany już obiekt klasy Pattern, np:

class NameAndSurnameValidator {

    private NameAndSurnameValidator() {
    }

    private static Pattern pattern = Pattern.compile("[^a-zA-ząćęłńóśźżĄĆĘŁŃÓŚŹŻ\\s]");

    private static Pattern capitalLetterPattern = Pattern.compile("[A-ZĄĆĘŁŃÓŚŹŻ]");

    public static void validate(String str) {
//more code...